tests/Application/Actions/Donations/CreateTest.php

```diff
@@ -988,6 +927,66 @@ public function testSuccessWithMinimalData(): void
         $this->assertEquals('123CampaignId12345', $payloadArray['donation']['projectId']);
     }
 
+    public function testCharityLackingCapabilities(): void
```

Comment:
From a quick look this test seems quite long and hard to follow - I feel like we should very rarely if ever want a test function that doesn't fit on the screen, and it might be overspecifying the test subject and making hard work for us in future if we want to change it. Happy to have a look at adjusting with you tomorrow if you like.

Reply:
I'm not sure it is feasible for this test to fit on one screen without extracting things specific to it, which IMO would make it more time consuming to understand. But I've now extracted out a bunch of common DI boilerplate I found was the same in a majority of tests, making a lot of them shorter.

Reply:
Thanks - it does look a lot clearer now, I think the AAA comments help quite a lot as well as the extractions you've done. I still feel like it should be feasible to make it simpler, reducing the use of the automocker and maybe either calling the controller under test directly instead of going through the app, or extracting the logic to a different class and testing that instead of testing the controller.